gribscan/gribscan.py

```diff
@@ -227,11 +232,26 @@ def get_time_offset(gribmessage, lean_towards="end"):
             offset += int(gribmessage["P1"]) * unit
         elif timeRangeIndicator == 1:
             pass
+        elif timeRangeIndicator == 3:
+            unit = time_range_units[
+                int(gribmessage.get("indicatorOfUnitOfTimeRange", 255))
+            ]
+            if lean_towards == "start":
+                offset += int(gribmessage["P1"]) * unit
+            elif lean_towards == "end":
+                offset += int(gribmessage["P2"]) * unit
         elif timeRangeIndicator == 10:
             unit = time_range_units[
                 int(gribmessage.get("indicatorOfUnitOfTimeRange", 255))
             ]
             offset += (int(gribmessage["P1"]) * 256 + int(gribmessage["P2"])) * unit
+        elif timeRangeIndicator == 123:
+            unit = time_range_units[
+                int(gribmessage.get("indicatorOfUnitOfTimeRange", 255))
+            ]
+            if lean_towards == "end":
+                N = int(gribmessage["numberIncludedInAverage"])
+                offset += N * int(gribmessage["P2"]) * unit
```

Comment:
I just reread the definition
> Average of N uninitialized analyses, starting at the reference time, at intervals of P2

I think the correct formula might actually be this one (e.g. if there are N things and the first is "reference time", then there are only (N - 1) deltas...)
```suggestion
                offset += (N - 1) * int(gribmessage["P2"]) * unit
```

Reply:
I tested the implementation using the one variable in ERA5 that is encoded with `timeRangeIndicator = 123` and the implementation as is leads to the following results:
```
lean_towards="start": 2000-01-01 00:00
lean_towards="end": 2000-02-01 00:00
```

So I would argue (or hope) that this is correct. Using `N -1` we would end at rather awkward values just before the end of the month.

We could also use the `cdo` approach and always lean towards start ;)
```
cgribexGetTsteptype: Time range indicator 123 unsupported, set to 0!
```

Reply:
TL;DR: I would argue that the current implementation is in line with the GRIB definition. However, other tools like cdo and cfgrib seem to prefer `lean_forward=start` (at least for this case).

Reply:
🤷‍♂️ IDK. Then let's keep it as is. The numbers will likely look nicer this way...